feat(spans): Adjust confidence calculations

wmak opened the pull request:
- Shuffles the variables around for the bounds calculations to match a discussion I had with @shellmayr
    - The resulting values didn't change significantly, I think part of the problem for timeseries is i'm still using the overall count instead of the count of the bucket, going to update that in a follow up
- Adds an option to change zscore
- Adds a function param to disable FPC

wmak committed:
feat(spans): Adjust bounds calculation

wmak committed:
ref: Add option for zscore, and param to disable fpc

wmak committed:
ref: round the bounds too

shellmayr commented on src/sentry/options/defaults.py:
```diff
@@ -1846,6 +1846,13 @@
 )
+# Used for the z-score when calculating the margin of error in performance
+register(
+    "performance.confidence.z-score",
```
The name of this option should probably contain the word extrapolation - so we can make sense of where this comes from.

shellmayr commented on src/sentry/search/events/datasets/spans_indexed.py:
```diff
@@ -792,14 +807,16 @@ def _resolve_margin_of_error(
             "multiply",
             [
                 # Based on a z score for a confidence level of 95%
```
nit: this comment has now lost its meaning

shellmayr commented on src/sentry/search/events/datasets/spans_indexed.py:
```diff
@@ -822,32 +839,46 @@ def _resolve_unadjusted_margin(
-        proportion_by_sample = Function("divide", [sampled_group, total_samples])
-        return Function(
+        proportion_by_sample = Function(
             "divide",
+            [
+                sampled_group,
+                Function(
+                    "multiply", [total_proportion, Function("avg", [Column("sampling_factor")])]
+                ),
+            ],
+            "proportion_by_sample",
+        )
+        return Function(
+            "round",
```
These functions are a bit hard to review due to the Snuba syntax - could you add some docstrings or similar with regular mathematical notation to indicate what certain functions are / should be calculating?

wmak committed:
ref: Address PR comments

- Renamed option to include extrapolation
- Added docstrings with formulas
- Removed outdated zscore comment

shellmayr reviewed (approved):
LGTM 👍
